core/src/main/java/com/cloud/storage/JavaStorageLayer.java

```diff
@@ -217,6 +229,19 @@ public boolean mkdirs(String path) {
         }
     }
 
+    public boolean isWorldReadable(File file) throws IOException {
+        Set<PosixFilePermission> permissions;
+        permissions = Files.getPosixFilePermissions(
+            Paths.get(file.getAbsolutePath()));
+
+        for (PosixFilePermission permission:permissions) {
+            if (permission.equals(PosixFilePermission.OTHERS_READ)) {
+                return true;
+            }
```

Comment:
no need of looping through all permissions, can simply check using permissions set contains `PosixFilePermission.OTHERS_READ` or not

```
if (permissions.contains(PosixFilePermission.OTHERS_READ)) {
                 return true;
}
```

Reply:
have simplified it.